pkg/ottl/parser_collection_test.go

```diff
@@ -0,0 +1,401 @@
+	}
+}
+
+func Test_EnableParserCollectionModifiedStatementLogging_False(t *testing.T) {
+	ps := mockParser(t, WithPathContextNames[any]([]string{"dummy"}))
+	core, observedLogs := observer.New(zap.InfoLevel)
+	telemetrySettings := componenttest.NewNopTelemetrySettings()
+	telemetrySettings.Logger = zap.New(core)
+
+	pc, err := NewParserCollection(
+		telemetrySettings,
+		WithParserCollectionContext("dummy", ps, newNopParsedStatementConverter[any]()),
+		EnableParserCollectionModifiedStatementLogging[any](false),
+	)
+	require.NoError(t, err)
+
+	_, err = pc.ParseStatementsWithContext("dummy", mockStatementsGetter{[]string{`set(attributes["foo"], "foo")`}}, true)
+	require.NoError(t, err)
+	require.Empty(t, observedLogs.TakeAll())
+}
+
+func Test_NopParsedStatementConverter(t *testing.T) {
+	type dummyContext struct{}
+
+	noop := newNopParsedStatementConverter[dummyContext]()
+	parsedStatements := []*Statement[dummyContext]{{}}
+	convertedStatements, err := noop(nil, nil, "", mockStatementsGetter{values: []string{}}, parsedStatements)
+
+	require.NoError(t, err)
+	require.NotNil(t, convertedStatements)
+	assert.Equal(t, parsedStatements, convertedStatements)
+}
+
+func Test_NewParserCollection_DefaultContextInferrer(t *testing.T) {
+	pc, err := NewParserCollection[any](componenttest.NewNopTelemetrySettings())
+	require.NoError(t, err)
+	require.NotNil(t, pc)
+	require.NotNil(t, pc.contextInferrer)
+}
+
+func Test_ParseStatements_Success(t *testing.T) {
+	ps := mockParser(t, WithPathContextNames[any]([]string{"foo"}))
+
+	pc, err := NewParserCollection(
+		component.TelemetrySettings{},
+		WithParserCollectionContext("foo", ps, newNopParsedStatementConverter[any]()),
+	)
+	require.NoError(t, err)
+	pc.contextInferrer = &mockStaticContextInferrer{"foo"}
+
+	statements := mockStatementsGetter{values: []string{`set(foo.attributes["bar"], "foo")`, `set(foo.attributes["bar"], "bar")`}}
+	result, err := pc.ParseStatements(statements)
+	require.NoError(t, err)
+
+	assert.IsType(t, []*Statement[any]{}, result)
+	assert.Len(t, result.([]*Statement[any]), 2)
+	assert.NotNil(t, result)
+}
+
+func Test_ParseStatements_MultipleContexts_Success(t *testing.T) {
+	fooParser := mockParser(t, WithPathContextNames[any]([]string{"foo"}))
+	barParser := mockParser(t, WithPathContextNames[any]([]string{"bar"}))
+	failingConverter := func(
+		_ *ParserCollection[any],
+		_ *Parser[any],
+		_ string,
+		_ StatementsGetter,
+		_ []*Statement[any],
+	) (any, error) {
+		return nil, errors.New("failing converter")
+	}
+
+	pc, err := NewParserCollection(
+		component.TelemetrySettings{},
+		WithParserCollectionContext("foo", fooParser, failingConverter),
+		WithParserCollectionContext("bar", barParser, newNopParsedStatementConverter[any]()),
+	)
+	require.NoError(t, err)
+	pc.contextInferrer = &mockStaticContextInferrer{"bar"}
+
+	statements := mockStatementsGetter{values: []string{`set(bar.attributes["bar"], "foo")`, `set(bar.attributes["bar"], "bar")`}}
```

Comment:
```suggestion
	// The `foo` context is never used, so these statements will successfully parse.
	statements := mockStatementsGetter{values: []string{`set(bar.attributes["bar"], "foo")`, `set(bar.attributes["bar"], "bar")`}}
```
Not strictly necessary, but I can't think of another way to explicitly capture the intent of this test (a failing parser that isn't called won't fail parsing).